src/state.rs

```diff
@@ -271,32 +271,75 @@ pub(crate) struct MultiProgressState {
 }
 
 impl MultiProgressState {
-    fn width(&self) -> usize {
```

Comment:
Please keep `width()` here to minimize changes in this large commit.

Reply:
Done!